src/x11/extensions/randr.cpp

```diff
@@ -108,55 +108,52 @@ namespace randr_util {
   vector<monitor_t> get_monitors(connection& conn, xcb_window_t root, bool connected_only, bool purge_clones) {
     vector<monitor_t> monitors;
 
-#if WITH_XRANDR_MONITORS
     if (check_monitor_support()) {
-      for (auto&& mon : conn.get_monitors(root, true).monitors()) {
+#if WITH_XRANDR_MONITORS
+      for (auto&& mon : conn.get_monitors(root, connected_only).monitors()) {
         try {
           auto name = conn.get_atom_name(mon.name).name();
-          monitors.emplace_back(make_monitor(XCB_NONE, move(name), mon.width, mon.height, mon.x, mon.y, mon.primary));
+          /* TODO: not reliable. The C++ binding does not expose
```

Comment:
Yeah, I couldn't figure out why xpp doesn't generate this. Though I really don't much experience with xpp internals.

We may not get around rewriting this whole loop using the C API

Reply:
Do you mean that we should get around by rewriting the whole loop using the C API? In this case, I can do that.

Reply:
Yes, that's what I meant. I didn't really formulate it directly enough :smile:

Reply:
OK, I have pushed that in a separate commit. I can squash them if you prefer. Be aware that I almost never do C++, so I have put auto everywhere and it compiles dans doesn't crash, so I think I am good.

Reply:
> OK, I have pushed that in a separate commit. I can squash them if you prefer.

No, this is fine. We can squash-merge the PR if things get out of hand.

> Be aware that I almost never do C++, so I have put auto everywhere and it compiles dans doesn't crash, so I think I am good.

Yeah, that works :smile: 

I think the code would be verbose if we had to explicitly add the types everywhere. And with xpp auto is required, because the types are impossible to write down.